application/src/main/java/org/togetherjava/tjbot/commands/moderation/temp/TemporaryModerationRoutine.java

```diff
@@ -81,13 +81,14 @@ private void checkExpiredActions() {
 
     private void processGroupedActions(@NotNull RevocationGroupIdentifier groupIdentifier) {
         // Do not revoke an action which was overwritten by a permanent action that was issued
-        // afterwards
+        // afterwards, or if the last action has not even expired yet
         // For example if a user was perm-banned after being temp-banned
         ActionRecord lastApplyAction = actionsStore
             .findLastActionAgainstTargetByType(groupIdentifier.guildId, groupIdentifier.targetId,
                     groupIdentifier.type)
             .orElseThrow();
```

Comment:
```suggestion
        // For example if a user was perm-banned after being temp-banned
        Instant actionExpiresAt = actionsStore
            .findLastActionAgainstTargetByType(groupIdentifier.guildId, groupIdentifier.targetId,
                    groupIdentifier.type)
            .orElseThrow()
            .actionExpiresAt();
            if (actionExpiresAt == null
                 || actionExpiresAt.isAfter(Instant.now())) {
```

Reply:
You are overlooking that its also used down below:

![usage](https://i.imgur.com/bCCNVKy.png)

Reply:
Oh, I missed that.
But maybe we can have a method in `ActionRecord` which checks whether it has expired, and then call that method itself. Since we need to check that more than once.

Reply:
Thats fair, I guess. Ill add such a method.

Reply:
done. please check again, thanks.